collectors/python.d.plugin/nvidia_smi/nvidia_smi.chart.py

```diff
@@ -326,6 +336,20 @@ def name(self):
     def full_name(self):
         return 'gpu{0} {1}'.format(self.num, self.name())
 
+    @handle_attr_error
+    def pci_link_gen(self):
+        return self.root.find('pci').find('pci_gpu_link_info').find('pcie_gen').find('max_link_gen').text
+
+    @handle_attr_error
+    def pci_link_width(self):
+        return self.root.find('pci').find('pci_gpu_link_info').find('link_widths').find('max_link_width').text.split('x')[0]
```

Comment:
Are you sure we need to use max and not current?

> **Current**
> The current link generation and width. These may be reduced when the GPU is not in use.

> **Maximum**
> The maximum link generation and width possible with this GPU and system configuration. For example, if the GPU supports a higher PCIe generation than the system.

Reply:
Yes, I want the maximum capable bandwidth which is constant whereas the current is meant to evolve accordingly to the GPU consumption.